src/main/java/com/atlassian/jira/cloud/jenkins/config/JiraCloudSiteConfig2.java

```diff
@@ -167,7 +178,19 @@ public FormValidation doTestConnection(
                 return FormValidation.error("Failed to retrieve secret");
             }
 
-            // TODO (ARC-1135): call a "ping" endpoint of the Jenkins app to test the connection
+            try {
+                boolean pingSuccess = pingApi.sendPing(webhookUrl, maybeSecret.get());
+                if (!pingSuccess) {
+                    return FormValidation.error(
+                            "Connection could not be established. Is the secret correct?");
+                }
```

Comment:
I've not seen the code on forge app where 'false' value is returned in case of incorrect secret. It is throwing an exception.

Reply:
@Harminder84 yeah, there is no case where `false` is returned. In that case, the app throws an exception, which is transformed into an HTTP 400 error that contains an error message, which will be shown here.